wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -537,6 +558,168 @@ instance BuildableSafeGen NewWallet where
         newwalName
         newwalOperation
 
+-- | Type for representation of extended public key in Base58-format.
+-- We use it for external wallets:
+-- 1. As a root PK to identify external wallet.
+-- 2. As a derived PK, please see 'AddressWithProof' type.
+newtype PublicKeyAsBase58 = PublicKeyAsBase58
```

Comment:
I would like to see the data consturctor marked as `Unsafe` or `Unchecked` and a smart constructor that takes a `PublicKey -> PublicKeyAsBase58` and does the hash.

Then, the arbitrary instance can rely on an arbitrary public key, and we get actually random encoded keys.

Base 58 can also instead be a ByteString for more efficient encoding, but that's unlikely to be a perf concern.